core/bootstrap/src/main/java/org/wildfly/swarm/bootstrap/util/TempFileManager.java

```diff
@@ -66,26 +70,43 @@ private void register(File file) {
         this.registered.add(file);
     }
 
-    public void close() {
-        for (File file : registered) {
-            deleteRecursively(file);
-        }
+    public synchronized void close() {
+        registered.forEach(TempFileManager::deleteRecursively);
+        registered.clear();
     }
 
     public static boolean deleteRecursively(File f) {
         if (!f.exists()) {
             return false;
         }
-        if (f.isDirectory()) {
-            File[] children = f.listFiles();
-            for (int i = 0; i < children.length; ++i) {
-                if (!deleteRecursively(children[i])) {
-                    return false;
+
```

Comment:
Switched the logic from a traditional recursion based search to a FileVisitor based approach.